packages/zend-xml/library/Zend/Xml/Security.php

```diff
@@ -45,6 +45,19 @@ protected static function heuristicScan($xml)
         }
     }
 
+    /**
+     * Wrapper for libxml_disable_entity_loader which is deprecated in PHP 8. libxml 2.9.0 disables external entity
+     * loading by default. PHP 8.0 and later has the option to enable it at runtime removed and libxml 2.9.0+ enforced.
```

Comment:
I found your sentence difficult to read due to weird language construct. perhaps this one is better?

```
     * loading by default. PHP 8.0 has deprecated libxml_disable_entity_loader() method and it requires libxml 2.9.0+.
```

Reply:
the link in same docblock said it's deprecated:
- https://github.com/php/php-src/pull/5867

Reply:
but the PR suggest checking for libxml instead:

```
if (LIBXML_VERSION < 20900) {
    libxml_disable_entity_loader();
}
```

- https://github.com/php/php-src/pull/5867#issuecomment-667941848

Reply:
PHP 8.0rc1 gives a deprecated warning indeed. That's what made me look into it.

I agree with the comment. English is not my first language.

I've noticed the libxml version check and used that at first.
However, after reading up I assumed that one can still execute `libxml_disable_entity_loader(false)` on PHP versions < 8 with LibXML version >= 2.9.0 to enable entity auto-loading globally. And that this could still be problematic.

While trying to look up documentation to show it to you I realized I lack(ed) proper understanding of what the function/workaround does! I'll dive into it to get a better understanding and reply again (and of course reconsider).

Thank you (all) for taking the time for an in-depth review.

Reply:
this is how I understand the problem:

- the whole problem is around disabling XXE attack vectors
- XXE can be disabled with libxml_disable_entity_loader(false), but it disables some more functionality, that could be useful (read discussion in linked php-src PR)
- libxml 2.9.0 disables XXE by default, so libxml_disable_entity_loader(false) call is not needed
- PHP 8 requires libxml 2.9.0 to be used (won't compile with older version)
- you can compile PHP < 8 with libxml 2.9.0
- PHP 8 adds deprecation to calls to libxml_disable_entity_loader(), altho not clear to me at this point is it noop call (doing nothing) or it still enables/disables something
- zend-xml calls libxml_disable_entity_loader() conditionally, so it must be okay to call or omit the call.

so, summarize that all: use libxml version check for all PHP versions

Reply:
You are right. According to https://github.com/php/php-src/pull/5867#issuecomment-664213053 calling `libxml_disable_entity_loader(false)` by itself doesn't do anything when compiled with libxml => 2.9.0, no matter the PHP version.

So the version check appears to be safe and the best option. I've verified this by testing with PHP 5.3.1, Libxml 2.7.3 and 5.6, 7.4 and 8 with 2.9.0+ using the script found here: https://gist.github.com/lukaskuzmiak/c8306a5af855c6faaaee . Only passing 'LIBXML_NOENT' will enable loading the external file on PHP with 2.9.0+.

To summarize:
- Revert the commit with the protected method name change (to keep changes to a minimum  for future syncing).
- Revert the removal of the unnecessary space to keep file formatting changes to a minimum.
- Replace the PHP version check with a libxml version check.
- Update the comment for the wrapper method to be less weird.

Thanks again for all the feedback. Since this is a first for me I'm unsure about the correct way to present the changes. Should I close this PR and start a new one? Or commit reverts and then the (hopefully) proper fix?

Reply:
I personally would rewrite the branch as intended and then force push new changes.

Reply:
@leendt you may also just push another commit(s) with changes, I'll eventually squash on merge to one commit anyway.

Reply:
Thanks for the compliments! I'm glad it shows that I'm putting quite some effort in understanding and getting it right. The in depth reviews were also quite motivating.

As for the old revisions, they'd become dangling commits after a rebase/rewrite, right? That's nice, still accessible yet automatically purged after a while.

I've reverted and committed the revert and the changes and pushed them. After that I've squashed them and force pushed the resulting commit because I was curious as to what it would do in this conversation view.

I was afraid GitHub might remove or move the conversation with a rebase + force push due to the branch rewriting. Glad to see GitHub handles it chronologically.

Reply:
I personally avoid referencing to commits that are not merged, especially the ones that do not exist (will not exist). you currently pushed commit (6d970a786378025680619649494699ec788d6bf4) message refers to rewritten commit, which will not exist once merged to master. i.e from local git clones. altho it will keep existing in GitHub UI.

perhaps you want to rewrite whole commit message? that you had commit that you removed is not interesting (useful) after this is merged to the default branch.

Reply:
Good point. I've reworded it.